server/atelier-queries.js

```diff
@@ -0,0 +1,116 @@
+    .catch(err => {
+      throw err;
+    });
+};
+
+// Returns the entire response as an object, including page, count, etc.
+const getProductReviews = (productId, headers) => {
+  return axios.get(`https://app-hrsei-api.herokuapp.com/api/fec2/hr-rpp/reviews?product_id=${productId}`, headers)
+    .then(result => result.data)
+    .catch(err => {
+      throw err;
+    });
+};
+
+// Returns the entire response as an object
+const getProductReviewsMeta = (productId, headers) => {
+  return axios.get(`https://app-hrsei-api.herokuapp.com/api/fec2/hr-rpp/reviews/meta?product_id=${productId}`, headers)
+    .then(result => result.data)
+    .catch(err => {
+      throw err;
+    });
+};
+
+// Returns the questions as an array
+const getProductQuestions = (productId, headers) => {
+  return axios.get(`https://app-hrsei-api.herokuapp.com/api/fec2/hr-rpp/qa/questions?product_id=${productId}`, headers)
+    .then(result => result.data.results)
+    .catch(err => {
+      throw err;
+    });
+};
+
+const getCurrentProductBundle = (productId, headers) => {
+  let currentProduct = {};
+  console.time('getCurrentProductBundle API Time');
+  const allCalls = [
+    getProductInfo(productId, headers),
+    getProductStyles(productId, headers),
+    getRelatedProducts(productId, headers),
+    getProductReviews(productId, headers),
+    getProductReviewsMeta(productId, headers),
+    getProductQuestions(productId, headers),
+  ];
+  return Promise.all(allCalls)
+    .then(responses => {
+      console.timeEnd('getCurrentProductBundle API Time');
+      return {
+        ...responses[0],
+        styles: [ ...responses[1] ],
+        related: [ ...responses[2] ],
+        reviews: { ...responses[3] },
+        reviewsMeta: { ...responses[4] },
+        questions: [ ...responses[5] ],
+      };
+    });
+};
+
+const getRelatedProductsBundle = (relatedProducts, headers) => {
+  console.time('getRelatedProductsBundle API Time');
+  const allRelatedProductCalls = [];
+  for (let i = 0; i < relatedProducts.length; i++) {
+    const relatedProduct = {};
+    const relatedProductCalls = [
+      getProductInfo(relatedProducts[i], headers),
+      getProductStyles(relatedProducts[i], headers),
+      getProductReviewsMeta(relatedProducts[i], headers)
+    ];
+    allRelatedProductCalls.push(Promise.all(relatedProductCalls)
+      .then(responses => {
+        return {
+          ...responses[0],
+          styles: [ ...responses[1] ],
+          reviewsMeta: { ...responses[2] },
+        };
+      })
+    );
+  }
+  return Promise.all(allRelatedProductCalls)
+    .then(responses => {
+      console.timeEnd('getRelatedProductsBundle API Time');
+      const allRelatedProductsBundle = {};
+      for (let i = 0; i < relatedProducts.length; i++) {
+        allRelatedProductsBundle[relatedProducts[i]] = responses[i];
+      }
+      return allRelatedProductsBundle;
+    });
+};
+
+module.exports = {
+  getCurrentProductBundle,
+  getRelatedProductsBundle
```

Comment:
I would consider also exporting:
getProductReviews
getProductReviewsMeta
and
getProductQuestions

...since those should be called after a user posts a new review or question, to update the page. If we only have the big getCurrentProductBundle, and that big hammer is the solution to all our nails, it's easy, but a little wasteful.

Reply:
Will add all these, good idea